evennia/utils/option.py

```diff
@@ -0,0 +1,62 @@
+from evennia.utils.utils import string_partial_matching
+
+
+class OptionHandler(object):
+    """
+    This is a generic Option handler meant for Typed Objects - anything that implements AttributeHandler.
+
+    It uses a dictionary to store-and-cache frequently used settings such as colors for borders or an
+    account's timezone.
+    """
+
+    def __init__(self, obj, options_dict=None, save_category=None):
+        if not options_dict:
+            options_dict = dict()
+        self.options_dict = options_dict
+        self.save_category = save_category
+        self.obj = obj
+        self.options = dict()
+
+    def __getitem__(self, item):
```

Comment:
I think it would be interesting to experiment with `__getattribute__` and `__setattr__` here instead, then you can do stuff like `.options.option_name` which makes for a much nicer API (cf `.db.attrname`). One must still keep access get/set methods for getting/setting options whose name you don't know at the time of coding of course. 

The drawback with messing with `__getattribute__`/`__setattr__` is that for the code inside this handler you need to be careful to avoid infinite loops (such as when getting things from `self`). The way around this is to make aliases `_GA = object.__getattribute__` and `_SA = object.__setattr__` and use those instead for the code inside the handler. It's a little tricky but it's no so much code in this case.

Reply:
This remains a thing. Being able to do `obj.options.optionname.value` makes sense for many common cases.